### PR TITLE
Add the definition of "directory" to the glossary

### PR DESCRIPTION
**Pull request by froggologies**
### Why:

Closes #18614

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Add these lines to line 207:
```yaml
- term: directory
  description: >-
    Computer files are cataloged in directories, which contain references to other computer files, and possibly other directories. The term "folder" is often used to describe directories on computers.
```

### Check off the following:

- [x] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

File: data/glossaries/external.yml [hunk 1]
```diff
@@ -204,6 +204,9 @@
     A diff is the difference in changes between two commits, or saved changes.
     The diff will visually describe what was added or removed from a file since
     its last commit.
+- term: directory
+  description: >-
+    Computer files are cataloged in directories, which contain references to other computer files, and possibly other directories. The term "folder" is often used to describe directories on computers.
 - term: dirty
```
Comment (HTC37) [inline data/glossaries/external.yml:210]: - term: directory
  description: >-
    Computer files are cataloged in directories, which contain references to other computer files, and possibly other directories. The term "folder" is often used to describe directories on computers.

Comment (HTC37) [reply data/glossaries/external.yml:210]: data/glossaries/external.yml

**Review by HTC37 — COMMENTED** [review]
data/glossaries/external.yml

**Comment by cmwilson21** [comment]
@FajarMaftuhFadli Thanks so much for opening a PR! I'll get this triaged for review :zap:

File: data/glossaries/external.yml [hunk 2]
```diff
@@ -204,6 +204,9 @@
     A diff is the difference in changes between two commits, or saved changes.
     The diff will visually describe what was added or removed from a file since
     its last commit.
+- term: directory
+  description: >-
+    Computer files are cataloged in directories, which contain references to other computer files, and possibly other directories. The term "folder" is often used to describe directories on computers.
```
Comment (meowius) [inline data/glossaries/external.yml:209]:
```suggestion
    A folder containing one or more files or folders. You can create directories to organize the contents of a repository.
```

**Review by meowius — COMMENTED** [review]
Thanks so much for opening this PR! I consulted the team, and we made some edits to align with the Docs style and to simplify language for newer users.

I'll go ahead and apply those changes for you, and we'll get this merged down! 🎉

**Review by meowius — APPROVED** [review]

**Pull request merged**